MCP Demo

TomHart opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no <!-- please update src/**/CHANGELOG.md files -->
| Docs?         | no <!-- required for new features -->
| Issues        | Demo (not working) code for MCP
| License       | MIT

<!--
Replace this notice by a description of your feature/bugfix.
This will help reviewers and should be a good start for the documentation.

Additionally (see https://symfony.com/releases):
 - Always add tests and ensure they pass.
 - For new features, provide some code snippets to help understand usage.
 - Features and deprecations must be submitted against branch main.
 - Update/add documentation as required (we can help!)
 - Changelog entry should follow https://symfony.com/doc/current/contributing/code/conventions.html#writing-a-changelog-entry
 - Never break backward compatibility (see https://symfony.com/bc).
-->

OskarStark reviewed (commented):
In the README we should document how you are able to use this server then

tom-hart-sky-uk commented:
@OskarStark Thanks for your feedback, I've addressed the comments and also fixed the code standards failure

chr-hertel commented on demo/README.md:
```diff
@@ -90,3 +90,21 @@ docker compose exec app bin/console app:blog:query
+### MCP
+Demo MCP server added with a `current-time` tool to return the current time, with an optional format string.
+
+To add the server, add the following to your `mcp.json` settings in your IDE:
```
```suggestion
To add the server, add the following configuration to your MCP Client's settings, e.g. your IDE:
```

TomHart commented:
> Thanks @TomHart for working on this - that was def missing 👍
> 
> I left a few comments, mostly minor stuff - please make sure to remove the security bundle - didn't comment on every changed caused by that change

Thanks @chr-hertel. I've addressed the comments you left

chr-hertel commented:
Looks good on the first glance - how did you test this? which the inspector?

let's add information about that so users can check it out themselves?

TomHart commented:
I used the CLI to paste in some JSON to list the tools. I've added the steps in the README. Is that what you were thinking, or something else?

OskarStark commented on demo/README.md:
```diff
@@ -90,3 +90,30 @@ docker compose exec app bin/console app:blog:query
 * You can reset the chat context by hitting the `Reset` button in the top right corner.
 * You find three different usage scenarios in the upper navbar.
+
+### MCP
```
```suggestion
### MCP

```

OskarStark commented on demo/README.md:
````diff
@@ -90,3 +90,30 @@ docker compose exec app bin/console app:blog:query
+}
+```
+
+#### Testing the MCP Server
````
```suggestion
#### Testing the MCP Server

```

OskarStark commented on demo/config/services.yaml:
```diff
@@ -18,3 +18,11 @@ services:
+    Symfony\AI\McpSdk\Capability\Tool\ToolExecutorInterface:
+        class: Symfony\AI\McpSdk\Capability\ToolChain
+        arguments:
+            - ['@App\MCP\Tools\CurrentTimeTool']
```
Off topic, but it would be nice if this is configurable via the bundle config

cc @Nyholm

TomHart replied:
Can you pass a service references in bundle config? E.g.

```yaml
mcp:
  tools:
    - '@App\MCP\Tools\CurrentTimeTool'
```

Would a tag + compiler pass perhaps be a cleaner way of doing this? 

```yaml
  App\MCP\Tools\CurrentTimeTool:
    tags:
      - { name: mcp_tool_executor }
      - { name: mcp_tool_metadata }
```

or just `name: mcp_tool`

chr-hertel reviewed (approved):
Looks good to me now :)

chr-hertel commented:
@TomHart please squash the commits to a single one

TomHart commented:
Done! Do you have any thoughts on https://github.com/symfony/ai/pull/132#discussion_r2217159246? I'd be happy to take a stab at implementing

chr-hertel commented:
> Do you have any thoughts on https://github.com/symfony/ai/pull/132#discussion_r2217159246? I'd be happy to take a stab at implementing

Hmm, on first thought, I think this would already be an improvement for now:
```yaml
mcp:
  tools:
    - '@App\MCP\Tools\CurrentTimeTool'
```

TomHart commented:
@chr-hertel I couldn't get it to work referencing services manually but added a compiler pass. Happy to give the list another go if you'd prefer https://github.com/symfony/ai/pull/168
